src/search/hnsw_indexer.h

```diff
@@ -92,7 +92,10 @@ struct HnswIndex {
   std::mt19937 generator;
   double m_level_normalization_factor;
 
-  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage);
+  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage,
+            std::random_device::result_type seed);
+  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage)
+      : HnswIndex(search_key, vector, storage, std::random_device()()) {}
```

Comment:
Maybe we can use a single constructor with a default argument? 
```
HnswIndex(const SearchKey& search_key, 
          HnswVectorFieldMetadata* vector, 
          engine::Storage* storage, 
          std::random_device::result_type seed = std::random_device()())
{
    ...
}
```

Reply:
Hi @Beihao-Zhou ,

I have updated related constructors. 😊

Best Regards,
Edward